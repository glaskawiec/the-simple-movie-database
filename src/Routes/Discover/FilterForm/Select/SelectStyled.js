import styled from 'styled-components';

const SelectStyled = styled.select`
display: inline-block;
  cursor: pointer;
  border-radius: 0.25rem;
  outline: 0;
  height: 2.5em;
  font-size: 1em;
  padding-left: 0.5em;
`;

export default SelectStyled;
