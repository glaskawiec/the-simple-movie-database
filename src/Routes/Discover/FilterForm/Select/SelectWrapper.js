import styled from 'styled-components';

const SelectWrapper = styled.div`
display: inline-block;
margin-right: 1.5em;
 margin-top: 1em;

  @media (max-width: 576px) {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    flex-wrap: wrap;
    width:80%;
    margin-left: auto;
    margin-right: auto;
  }

`;

export default SelectWrapper;
