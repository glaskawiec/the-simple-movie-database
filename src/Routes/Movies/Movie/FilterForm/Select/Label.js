import styled from 'styled-components';

const Label = styled.label`
  display: block;
  margin-bottom: 4px;
  font-size: 1em;                      
`;

export default Label;
