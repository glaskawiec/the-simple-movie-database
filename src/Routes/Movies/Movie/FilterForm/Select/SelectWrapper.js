import styled from 'styled-components';

const SelectWrapper = styled.div`
display: inline-block;
margin-right: 1.5em;

&:last-child {
margin-right: 0;
}

`;

export default SelectWrapper;
