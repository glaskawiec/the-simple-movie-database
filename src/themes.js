const Themes = {
  light: {
    colors: {
      primary: '#F2355B',
      secondary: '#161616',
      background: '#FFFFFF',
      text: '#000000',
      mobileMenuOverlay: '#000000',
      card: '#FFFFFF',
      themeToggleThumb: '#161616',
      paginationHover: '#CDCDCD',
    },
  },
  dark: {
    colors: {
      primary: '#F2355B',
      secondary: '#090909',
      background: '#161616',
      text: 'hsla(0,0%,100%,0.88)',
      mobileMenuOverlay: '#FFFFFF',
      card: '#090909',
      themeToggleThumb: '#FFFFFF',
      paginationHover: '#292929',
    },
  },
};

export default Themes;
