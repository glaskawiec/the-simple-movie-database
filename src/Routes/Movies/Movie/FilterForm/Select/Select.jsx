import React from 'react';
import PropTypes from 'prop-types';
import SelectStyled from './SelectStyled';
import Label from './Label';
import SelectWrapper from './SelectWrapper';

const Select = ({ label, options }) => (
  <SelectWrapper>
    <Label>{label}</Label>
    <SelectStyled>
      {options.map(option => (<option key={option}>{option}</option>))}
    </SelectStyled>
  </SelectWrapper>
);

Select.propTypes = {
  label: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Select;
