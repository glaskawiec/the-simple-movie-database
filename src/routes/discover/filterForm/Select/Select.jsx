import React from 'react';
import PropTypes from 'prop-types';
import SelectStyled from './SelectStyled';
import Label from './Label';
import SelectWrapper from './SelectWrapper';

const Select = ({
  label, options, onChange, value,
}) => (
  <SelectWrapper>
    <Label>{label}</Label>
    <SelectStyled value={value} onChange={onChange}>
      {options.map(({ value: optionValue, label: optionLabel }) => (
        <option value={optionValue} key={optionLabel}>{optionLabel}</option>
      ))}
    </SelectStyled>
  </SelectWrapper>
);

Select.propTypes = {
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Select;
