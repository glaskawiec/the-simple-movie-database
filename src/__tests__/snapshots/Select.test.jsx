import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import themes from '../../themes';
import Select from '../../common/select/Select';

test('Select snapshot', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={themes.light}>
        <Select
          label="snap"
          onChange={jest.fn()}
          options={[{ label: 'snap', value: 'snap' }]}
          value="snap"
        />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
