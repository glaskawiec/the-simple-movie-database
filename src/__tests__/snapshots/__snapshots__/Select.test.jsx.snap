// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select snapshot 1`] = `
<div
  className="sc-htpNat hkVWJd"
>
  <label
    className="sc-bwzfXH gWEriG"
  >
    snap
  </label>
  <select
    className="sc-bdVaJa jVkQAe"
    onChange={[MockFunction]}
    value="snap"
  >
    <option
      value="snap"
    >
      snap
    </option>
  </select>
</div>
`;
